ocaml/idl/datamodel_utils.ml

```diff
@@ -579,6 +587,34 @@ let messages_of_obj (x : obj) document_order : message list =
     ; msg_tag= FromObject GetByUuid
     }
   in
+  (* Get by UUID - Doesn't raise an exception, returns an option instead *)
+  let uuid_opt =
```

Comment:
We don't really need this in the datamodel (_types and _utils) until we want to expose it in the API. I would keep in restricted to the Db_actions for now to avoid unintended side-effects. We can generate code under the `GetByUuid` case in gen_db_actions.ml.